### PR TITLE
[DOCS] Will/013 cleanup

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
- Pandas/filesystem Datasource how-to guide: replace the "{base_directory}" var with actual path
- Relink the note that explains how to choose a connector to guides/how_to_guides/configuring_datasources/how_to_choose_which_data_connector_to_use.html in all how-to guides where it appears
- Say "create a new jupyter notebook" before the first code snippet in all Datasources how-to guides
- Yaml files in Datasources how-to guides use 2 rather than 4 spaces for compatibility with existing great_expectations.yml
- Add "Step 5" that explains how to copy the yml snippet to all Datasources how-to guides
- various typos

### CODE REVIEW

File: docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_snowflake_datasource.rst [hunk 1]
```diff
@@ -397,10 +404,10 @@ Additional Notes
                 class_name: SimpleSqlalchemyDatasource
                 credentials:
                     drivername: snowflake
-                    username: ${snowflake_username}
```
Comment (eugmandel) [inline docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_snowflake_datasource.rst:400]: I think we should either keep the variable substitution syntax (${snowflake_username}) or something like "YOUR_SNOWFLAKE_USERNAME_HERE". The brackets make it something in between.

**Review by eugmandel — CHANGES_REQUESTED** [review]
Looks great! One comment re the variables - please see inline.

File: docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_snowflake_datasource.rst (hunk 1 above)
Comment (Shinnnyshinshin) [reply docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_snowflake_datasource.rst:400]: Great point. Will update now :)

Comment (Shinnnyshinshin) [reply docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_snowflake_datasource.rst:400]: I have a preference for YOUR_SNOWFLAKE_USERNAME_HERE <~

Comment (Shinnnyshinshin) [reply docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_snowflake_datasource.rst:400]: I tried the '<user_name>' format, and it made things more confusing, especially with MSSQL connection string

Comment (Shinnnyshinshin) [reply docs/guides/how_to_guides/configuring_datasources/how_to_configure_a_snowflake_datasource.rst:400]: now the credentials are represented in `YOUR_SNOWFLAKE_USERNAME`-like format

**Review by eugmandel — APPROVED** [review]
Perfect - let's merge!

**Pull request merged**